frontend/pkg/frontend/frontend.go

```diff
@@ -237,6 +237,80 @@ func (f *Frontend) ArmResourceRead(writer http.ResponseWriter, request *http.Req
 	writer.WriteHeader(http.StatusOK)
 }
 
+func (f *Frontend) GetNodePoolOfClusterByName(writer http.ResponseWriter, request *http.Request) {
+	ctx := request.Context()
+
+	versionedInterface, err := VersionFromContext(ctx)
+	if err != nil {
+		f.logger.Error(err.Error())
+		arm.WriteInternalServerError(writer)
+		return
+	}
+
+	f.logger.Info(fmt.Sprintf("%s: GetNodePools", versionedInterface))
+
+	// URL path is already lowercased by middleware.
+	resourceID := request.URL.Path
+	subscriptionID := request.PathValue(PathSegmentSubscriptionID)
+	clusterDoc, err := f.dbClient.GetClusterDoc(ctx, resourceID, subscriptionID)
+	if err != nil {
+		if errors.Is(err, database.ErrNotFound) {
+			f.logger.Error(fmt.Sprintf("existing cluster document not found for cluster: %s on GET node pool by name", resourceID))
+			writer.WriteHeader(http.StatusNoContent)
+			return
+		} else {
+			f.logger.Error(err.Error())
+			arm.WriteInternalServerError(writer)
+			return
+		}
+	}
+
+	nodePoolDoc, err := f.dbClient.GetNodePoolDoc(ctx, resourceID)
+	if err != nil {
+		if errors.Is(err, database.ErrNotFound) {
+			f.logger.Error(fmt.Sprintf("existing node pool document not found for node pool: %s on GET node pool by name", resourceID))
+			writer.WriteHeader(http.StatusNoContent)
+			return
+		}
+
+		f.logger.Error(err.Error())
+		arm.WriteInternalServerError(writer)
+		return
+	}
+
+	targetNodePoolName := nodePoolDoc.ID
+	// nodePoolGetResponse, err := f.conn.ClustersMgmt().V1().Clusters().Cluster(clusterDoc.ClusterID).NodePools().NodePool(targetNodePoolName).Get().Send()
+	nodePoolGetResponse, err := f.conn.ClustersMgmt().V1().Clusters().Cluster(clusterDoc.ClusterID).NodePools().NodePool(targetNodePoolName).Get().Send()
```

Comment:
Not an ask for this pull request, but I wonder if there would be benefit in defining a CSClient interface similar to our DBClient interface, so that Cluster Service could be mocked for testing.

Reply:
Yes! We agreed in a previous [comment](https://github.com/Azure/ARO-HCP/pull/164#discussion_r1628086699) of a PR that it would be beneficial 🙂